core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -116,49 +124,52 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    */
   public static class SSHActionConfig extends PluginConfig {
     @Description("Command to be executed on the remote host. Should include filepath of script and any arguments")
+    @Macro
     private String command;
 
     @Description("Host name of the remote machine where the command needs to be executed.")
+    @Macro
     private String host;
 
     @Nullable
     @Description("Port to connect to. Defaults to 22")
+    @Macro
     private Integer port;
 
     @Description("User name used to connect to host")
+    @Macro
     private String user;
 
-    @Description("File path to Private key")
-    private String privateKeyFile;
+    @Description("Key used to lookup private key file from secure storage")
+    @Macro
+    private String privateKeyFileLookupKey;
 
     @Nullable
-    @Description("Password associated with private key")
-    private String password;
+    @Description("Passphrase associated with private key")
+    @Macro
+    private String passphrase;
 
     @Nullable
     @Description("Variable used to hold desired key name to be associated with script output in the instance " +
```

Comment:
From my understanding it's meant to be exposed to users to (almost exclusively) substitute plugin properties with a macro ```${output}``` where "output" is the key they specified. I agree it could use some work. Will try to update it.